### PR TITLE
Code coverage

### PR DESCRIPTION
**Pull request by tommysullivan**

### CODE REVIEW

**Comment by tommysullivan** [comment]
@bob-bins looks like i can run and get coverage report locally, including if i add reporter "html" in the "nyc" block of package.json

it looks like the pull request job is trying to update codecov.io with the report, but does not find a report; probably because the report gets generated within the running docker container during the docker build.

so i was thinking of mounting the coverage folder in the calling environment so that this step could see the coverage file. hopefully if it does then our viewer here at https://codecov.io/gh/heartpoints/heartpoints.org will actually show something.

File: src/cicd/brew.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+source "src/cicd/reflect.sh"
```
Comment (bob-bins) [inline src/cicd/brew.sh]: i hope there are no circular dependencies! :P

File: README.md [hunk 2]
```diff
@@ -7,13 +7,18 @@ Welcome to the Heartpoints.org website repository!
 
 - [Initial Development Environment Setup](docs/initialDeveloperSetup.md)
 - [Using Github](docs/usingGithub.md) - Issues, Pull Requests, and Source Control
-- [Heartpoints CLI](docs/cli.md) - Overview for the `./hp` command line interface for running dev tasks (ie: compiling/testing)
+- [Heartpoints CLI](docs/cli.md) - Overview for the `./hp` command line interface for running dev tasks (ie:
```
Comment (bob-bins) [inline README.md]:
```suggestion
- [Heartpoints CLI](docs/cli.md) - Overview for the `./hp` command line interface for running dev tasks (e.g.: 
```

File: docs/codeCoverage.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+4. IDE Plugin - A plugin that can read (and continuously keep an eye on changes to) the coverage Report, updating
+   the UI of the IDE to visually and easily show what code is tested vs. untested.
+5. CICD Automation - A means to produce the report and send it to github for inline-viewing during code review
+6. Github UI - A means to view code coverage inline on github.com during code review (may require browser plugin)
+
+# Key Inputs
+
+1. Tests currently run using ts-mocha, which helps automatically compile and run mocha
+2. Instrumenting needs to occur somewhere between reading the raw ts / tsx source and running the tests
+3. Sourcemaps may need to be produced, as well as used, to create a coverage report targetting original source
```
Comment (bob-bins) [inline docs/codeCoverage.md]:
```suggestion
3. Sourcemaps may need to be produced, as well as used, to create a coverage report targeting original source
```

File: docs/codeCoverage.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+   (and not the compiled javascript that comes out of it and is in fact what gets tested / executed)
+
+# Possible Solutions
+
+Are there any solutions that show results in github.com without requiring a browser plugin? Haven't found any yet.
```
Comment (bob-bins) [inline docs/codeCoverage.md]: Side note, but i think Gitlab may have something like this

File: docs/tdd.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+Test Driven Development
+-----------------------
+
+We use Test Driven Development (TDD) to get our work under automated test. Please read up on the following materials
+to undewrstand more about TDD in practice:
```
Comment (bob-bins) [inline docs/tdd.md]:
```suggestion
to understand more about TDD in practice:
```

**Pull request closed without merging**